L1Trigger/TrackerDTC/test/AnalyzerDAQ.cc

```diff
@@ -0,0 +1,132 @@
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "L1Trigger/TrackerDTC/interface/Setup.h"
+
+#include <TProfile.h>
+#include <TH1F.h>
+
+#include <vector>
+#include <deque>
+
+using namespace std;
+using namespace edm;
+using namespace trackerDTC;
+
+namespace trackerDTC {
+
+  /*! \class  trackerDTC::AnalyzerDAQ
+   *  \brief  Class to analyze TTCluster Occupancies on DTCs
+   *  \author Thomas Schuh
+   *  \date   2020, Oct
+   */
+  class AnalyzerDAQ : public one::EDAnalyzer<one::WatchRuns, one::SharedResources> {
+  public:
+    AnalyzerDAQ(const ParameterSet& iConfig);
+    void beginJob() override {}
+    void beginRun(const Run& iEvent, const EventSetup& iSetup) override;
+    void analyze(const Event& iEvent, const EventSetup& iSetup) override;
+    void endRun(const Run& iEvent, const EventSetup& iSetup) override {}
+    void endJob() override {}
+
+  private:
+
+    // ED input token of accepted TTClusters
+    EDGetTokenT<TTClusterDetSetVec> edGetToken_;
+    // Setup token
+    ESGetToken<Setup, SetupRcd> esGetTokenSetup_;
+    // stores, calculates and provides run-time constants
+    const Setup* setup_;
+
+    // Histograms
+
+    TProfile* profModules_;
+    TH1F* hisModules_;
+    TProfile* profDTCs_;
+    TH1F* hisDTCs_;
+    TH1F* hisTracker_;
+  };
+
+  AnalyzerDAQ::AnalyzerDAQ(const ParameterSet& iConfig) {
+    usesResource("TFileService");
+    // book input ED products
+    const auto& inputTag = iConfig.getParameter<InputTag>("InputTagTTClusterDetSetVec");
+    edGetToken_ = consumes<TTClusterDetSetVec>(inputTag);
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    // initial ES products
+    setup_ = nullptr;
+  }
+
+  void AnalyzerDAQ::beginRun(const Run& iEvent, const EventSetup& iSetup) {
+    // helper class to store configurations
+    setup_ = &iSetup.getData(esGetTokenSetup_);
+    // book histograms
+    Service<TFileService> fs;
+    TFileDirectory dir;
+    dir = fs->mkdir("Modules");
+    int maxOcc = 150;
+    int numChannels = setup_->numDTCs() * setup_->numModulesPerDTC();
+    hisModules_ = dir.make<TH1F>("His Module Occupancy", ";", maxOcc, -.5, maxOcc - .5);
+    profModules_ = dir.make<TProfile>("Prof Module Occupancy", ";", numChannels, -.5, numChannels - .5);
+    dir = fs->mkdir("DTCs");
+    maxOcc = 3456;
+    numChannels = setup_->numDTCs();
+    hisDTCs_ = dir.make<TH1F>("His DTC Occupancy", ";", maxOcc / 16, -.5, maxOcc - .5);
+    profDTCs_ = dir.make<TProfile>("Prof DTC Occupancy", ";", numChannels, -.5, numChannels - .5);
```

Comment:
For profModules & profDTCs, the meaning of the x-axis is unclear. i.e. Don't know if DTC 23 or module 98 are in the inner or outer Tracker layer. Any way to clarify this?

Reply:
no

Reply:
So the module and DTC numbers being plotted have no physical meaning?

Reply:
dtcId / 24 is the nonant number. (dtcId % 24) / 12 is the z side (something like 0 = - and 1 = + or vice versa). (dtcId % 12) / 6 gives you 0=PS and 1=2S I believe. Within the PS portion one can distinguish between the first half = 10g and the second half = 5g I believe as well.

Reply:
Minimal improvement would be to add comment, where these two histos are booked, saying "dtcId / 24 is the nonant number. (dtcId % 24) / 12 is the z side (something like 0 = - and 1 = + or vice versa). (dtcId % 12) / 6 gives you 0=PS and 1=2S".